src/Web/Controllers/AccountController.cs

```diff
@@ -73,11 +86,38 @@ public async Task<IActionResult> SignIn(LoginViewModel model, string returnUrl =
             }
             ViewData["ReturnUrl"] = returnUrl;
 
-            var result = await _signInManager.PasswordSignInAsync(model.Email, model.Password, model.RememberMe, lockoutOnFailure: false);
-            if (result.Succeeded)
+            ApplicationUser applicationUser = new ApplicationUser()
+            {
+                UserName = model.Email
+            };
+            var passwordHash = _passwordHasher.HashPassword(applicationUser, model.Password);
+
+            var deviceContext = new DeviceContext
```

Comment:
Since the device context is not otherwise used, I would just inline the IP address value on the sign in request object